//   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
//     University of Wisconsinâ€”Madison.
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.

syntax = "proto2";

package quickstep.serialization;

message BloomFilter {
  // The default values were determined from empirical experiments.
  // These values control the amount of false positivity that
  // is expected from Bloom Filter.
  // - Default seed for initializing family of hashes = 0xA5A5A5A55A5A5A5A.
  // - Default bloom filter size = 10 KB.
  // - Default number of hash functions used in bloom filter = 5.
  optional fixed64 bloom_filter_seed = 1 [default = 0xA5A5A5A55A5A5A5A];
  optional uint32 bloom_filter_size = 2 [default = 10000];
  optional uint32 number_of_hashes = 3 [default = 5];
}
